File: SAMPLES/PHP/control.php
<?php
// ============================================================
// control.php -- Structures de controle pour PHP86
// if/elseif/else, while, for, foreach, switch, break, continue
// ============================================================

echo "=== control.php ===\n";

// --- If / elseif / else ---
$n = 42;
echo "n = ", $n, "\n";
if ($n > 100) {
    echo "grand\n";
} elseif ($n > 10) {
    echo "moyen\n";
} else {
    echo "petit\n";
}

// --- While ---
echo "while 1..5 : ";
$i = 1;
while ($i <= 5) {
    echo $i, " ";
    $i = $i + 1;
}
echo "\n";

// --- Do-While ---
echo "do-while 3..1 : ";
$j = 3;
do {
    echo $j, " ";
    $j = $j - 1;
} while ($j > 0);
echo "\n";

// --- For ---
echo "for 0..4 : ";
for ($k = 0; $k < 5; $k++) {
    echo $k, " ";
}
echo "\n";

// --- For avec break ---
echo "for break a 3 : ";
for ($m = 0; $m < 10; $m++) {
    if ($m == 3) {
        break;
    }
    echo $m, " ";
}
echo "\n";

// --- For avec continue ---
echo "for impairs 0..9 : ";
for ($p = 0; $p < 10; $p++) {
    if ($p % 2 == 0) {
        continue;
    }
    echo $p, " ";
}
echo "\n";

// --- Switch ---
$code = 2;
echo "switch(", $code, ") : ";
switch ($code) {
    case 1:
        echo "un\n";
        break;
    case 2:
        echo "deux\n";
        break;
    case 3:
        echo "trois\n";
        break;
    default:
        echo "autre\n";
}

// --- Operateurs logiques ---
$a = 1;
$b = 0;
echo "a && b = ", ($a && $b), "\n";
echo "a || b = ", ($a || $b), "\n";
echo "!a     = ", (!$a), "\n";

echo "=== Fin control.php ===\n";

File: SAMPLES/RUST/control.rs
// control.rs - Structures de controle: if/else, while, match
fn classify(x: i32) -> i32 {
    if x < 0 { return 0; }
    else if x == 0 { return 1; }
    else { return 2; }
}

fn sum_to(n: i32) -> i32 {
    let mut acc: i32 = 0;
    let mut i: i32 = 0;
    while i < n { acc = acc + i; i = i + 1; }
    return acc;
}

fn label_of(code: i32) -> i32 {
    match code {
        0 => { return 10; }
        1 => { return 20; }
        _ => { return 99; }
    }
    return 0;
}

pub fn main() {
    let mut r: i32 = 0;
    r = classify(0 - 5);
    println!("classify(-5) = {}", r);
    r = classify(0);
    println!("classify(0) = {}", r);
    r = classify(42);
    println!("classify(42) = {}", r);
    r = sum_to(10);
    println!("sum_to(10) = {}", r);
    r = label_of(0);
    println!("label_of(0) = {}", r);
    r = label_of(1);
    println!("label_of(1) = {}", r);
    r = label_of(5);
    println!("label_of(5) = {}", r);
    debug_assert!(r);
}
